c-api/compat-5.3.c

```diff
@@ -15,6 +15,25 @@
 /* definitions for Lua 5.1 only */
 #if defined(LUA_VERSION_NUM) && LUA_VERSION_NUM == 501
 
+#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (!defined (__MINGW32__) && defined(__GNUC__) && (__GNUC__ < 6))
```

Comment:
Put the #if inside of the `#ifndef`

Reply:
I'm confused: which part of the `#if` goes inside the `#ifndef` ?

Reply:
You want users to be able to pass `-DHAVE_STRERROR_R=1` to their compiler.
So you end up with:
```C
#ifndef COMPAT53_HAVE_STRERROR_R
#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (!defined (__MINGW32__) && defined(__GNUC__) && (__GNUC__ < 6))
#define COMPAT53_HAVE_STRERROR_R 1
#else
#define COMPAT53_HAVE_STRERROR_R 0
#endif
#endif
```

Which is verbose due to the new GCC7 `-Wexpansion-to-defined` warning.